File: 3002.cc
#include <stdio.h>
#include <string.h>
#include <stack>
#include <vector>
#include <queue>
#include <iostream>
#include <algorithm>
#define LL long long int
using namespace std;
long long  gcd(long long  a,long long  b)
{
   return b==0?a:gcd(b,a%b);
}
int shuru()
{
    int res = 0, flag = 0;
    char ch;
    if((ch = getchar()) == '-') flag = 1;
    else if(ch >= '0' && ch <= '9') res = ch - '0';
    while((ch = getchar()) >= '0' && ch <= '9')
        res = res * 10 + (ch - '0');
    return flag ? -res : res;
}
int main()
{
    int t;
    t=shuru();
    while(t--)
    {
        bool flag=false;
        int n;
        int m;
        n=shuru();
        long long k=1;
        for(int i=0;i<n;i++)
        {
           m=shuru();
            if(!flag) k=k*m/gcd(k,m);
            if(k>1000000000) flag=true;
        }
         if(flag) puts("More than a billion.");
        else printf("%lld\n",k);
    }
    return 0;
}

File: 3002.java
/* @author: */
import java.util.Scanner;
import java.util.Arrays;
public class Main{

 static long gcd(long a,long b)
{
  if(b==0) return a;
  return gcd(b,a%b);
}


 public static void main(String args[])
{
 Scanner sc=new Scanner(System.in);
 int t,n,i;
 long p[]=new long[6];
 t=sc.nextInt();
 while((t--)!=0)
 {
    n=sc.nextInt();
    for(i=0;i< n;i++)
      p[i]=sc.nextLong();
    long u=p[0];
    boolean bb=false;
    for(i=1;i< n;i++)
    {
	u=u*p[i]/gcd(p[i],u);
	if(u>1000000000){
         bb=true;
	  break;
	}
    }
    if(!bb) System.out.printf("%d\n",u);
    else System.out.println("More than a billion.");
   }
  }
}

